lib/dev_server.dart

```diff
@@ -84,6 +93,30 @@ class ButterflyDevServer {
     request.response.write(JSON.encode(result));
   }
 
+  Future<Null> _serveDevServerClientFile(HttpRequest request) async {
+    final packagesFile = new File('.packages');
+
+    if (!await packagesFile.exists()) {
+      throw new StateError('.packages file not found. Have you run pub get?');
+    }
+
+    List<String> packageInfoParts = await packagesFile
+        .openRead()
+        .transform(const Utf8Decoder())
+        .transform(const LineSplitter())
+        .map((String line) => line.split(':'))
+        .firstWhere((List<String> parts) => parts[0] == 'butterfly',
+            defaultValue: () => null);
+
+    if (packageInfoParts == null) {
+      throw new StateError('"butterfly" package not found in .packages file. Please check your pubspec.yaml and run pub get again.');
```

Comment:
Don't throw `Error` unless you want to communicate consumers to gracefully handle program ending. Looks like that is what will happen here anyways. Do you need a more specialized error? Your own?

I don't know what your max line length is but consider taking advantage of Dart's string concatenation across different lines.